### DISCUSSION TITLE
Electron-vite and Prisma arranged me a challenging weekend.

### DISCUSSION THREAD

**Discussion by Evilu** [prisma/prisma#21027] [Q&A]
### Question

So as you may have guessed from the title, I've had a challenging few days.
Prisma is a very easy to use library that works almost by itself, but when you package it with electron-builder things can get complicated, and when you add Vite it gets even  more confusing.
Add to that the fact that there is no fast way that i know of to debug a built electron app (out of prints in the renderer or write files) and you've got a few days of trial and error
I hope that this message will help to anyone who will encounter difficulties in the future, and maybe it'll give some ideas to contributors how to develop an easier integration.

**Used versions:**
```
   "electron-vite": "^1.0.21",
    "prisma": "^5.0.0",
    "@prisma/client": "^5.0.0"

```
**Intended environment:** 
Windows 11

**Intended Database:** 
MySQL 8.0

As said: Prisma is usually very easy to use, you develop on one computer, push to production and wait for it to go to production with` Prisma Generate `Command.

Everything worked perfectly on dev environment. 
Problem started when i built the app using electron-vite's `build:win` command.
Tried installing the app on windows 11, and got an error from the main process: 
` Cannot find module'@prisma/client'`

Tired many different approaches suggested here, what worked for me is adding:
```
 "build": {
    "extraResources": [
      "prisma/**/*",
      "node_modules/.prisma/**/*",
      "node_modules/@prisma/client/**/*"
    ]
  }
```
  to the `package.json` file.
  
 it means that Prisma cannot be compiled into the `electron.asar` file (whenever you're using bytecode plugin or not) and needs to stay as is in the `resources` folder on target.
  
 Ok, i've been expecting everything to work now, but it didn't.
 guess what, environment files:
 Electron-Vite using its own approach for env files: 
 [https://electron-vite.org/guide/env-and-mode.html#global-env-variables](url)
 And as we know, Prisma is also using the `schema.prisma` file which depends on your .env file.
 
```
datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}
```

Anyway i've solved it with adding a prefix for main process environment variables to the `electron.vite.config.js `file as electron-vite requires.

```
export default defineConfig({
  main: {
    envPrefix: 'M_VITE_'
  }
  // ...
})
```
Now i need to add the url environment variable when creating the PrismaClient like this:
```
const prisma = new PrismaClient({
  datasources: {
    db: {
      url: import.meta.env.M_VITE_DATABASE_URL
    }
  }
});
```

Ok, now it worked well as a built app on my mac, but still did nothing on windows.

Well guess what, when you hit the` Prisma Generate `Command, Prisma is built according to your operating system, it didn't worked for me because i've built it on a darwin-arm64 mac, and prisma has system specific executables as dependencies. 
[https://www.prisma.io/docs/concepts/components/prisma-schema/generators](url)

What you need to do is go back to your `schema.prisma` file and explicitly note your intended os.
for me it was:

```
generator client {
  provider = "prisma-client-js"
  binaryTargets = ["windows"]
}
```
 Read the api docs for other options:

[https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#binarytargets-options](url)

Hope it'll help you guys.

**Comment by ludralph** [comment]
Hi @Evilu 👋 

Thank you for sharing your solution. I'm certain the community will find this really helpful.

**Comment by Stormpass** [comment]
Hi， @Evilu  
Thanks a lot, It is really helpful. It is work for me
my project include Bytenode、asar、rollup tools

**Comment by WebAFilippov** [comment]
did you use the path to the database deleted on the server in env?

**Comment by efeliza00** [comment]
Does this works on electron-forge?
